samples/sensor/ltc4162/src/main.c

```diff
@@ -0,0 +1,271 @@
+			       &input_current) < 0) {
+		printk("Can't read LTC4162 Charger Input current\n");
+		return;
+	}
+
+	ltc4162_show_values("Input Current in Amps", input_current);
+
+	if (sensor_sample_fetch_chan(dev,
+			SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_CURRENT) < 0) {
+		printk("Failed to fetch constant charge current\n");
+		return;
+	}
+
+	if (sensor_channel_get(dev,
+			SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_CURRENT, &const_chrg_current) < 0) {
+		printk("Can't read LTC4162 constant charge current\n");
+	}
+
+	ltc4162_show_values("Constant current in Amps", const_chrg_current);
+
+	if (sensor_sample_fetch_chan(dev,
+			SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_CURRENT_MAX) < 0) {
+		printk("Failed to fetch constant charge current max\n");
+		return;
+	}
+
+	if (sensor_channel_get(dev, SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_CURRENT_MAX,
+			       &const_chrg_current_max) < 0) {
+		printk("Can't read LTC4162 constant charge current max\n");
+		return;
+	}
+
+	ltc4162_show_values("Constant current max in Amps", const_chrg_current_max);
+
+
+	if (sensor_sample_fetch_chan(dev,
+			SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_VOLTAGE) < 0) {
+		printk("Failed to fetch constant charge voltage\n");
+		return;
+	}
+
+	if (sensor_channel_get(dev, SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_VOLTAGE,
+			       &const_chrg_voltage) < 0) {
+		printk("Can't read LTC4162 const charge voltage\n");
+		return;
+	}
+
+	ltc4162_show_values("Constant voltage in Volt", const_chrg_voltage);
+
+	if (sensor_sample_fetch_chan(dev,
+			SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_VOLTAGE_MAX) < 0) {
+		printk("Failed to fetch constant charge voltage max\n");
+		return;
+	}
+
+	if (sensor_channel_get(dev,
+			       SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_VOLTAGE_MAX,
+			       &const_chrg_voltage_max) < 0) {
+		printk("Can't read LTC4162 const charge voltage max\n");
+	}
+
+	ltc4162_show_values("Constant voltage max in Volt", const_chrg_voltage_max);
+
+	if (sensor_sample_fetch_chan(dev,
+			SENSOR_CHAN_CHARGER_INPUT_CURRENT_LIMIT) < 0) {
+		printk("Failed to fetch input current limit\n");
+		return;
+	}
+
+	if (sensor_channel_get(dev, SENSOR_CHAN_CHARGER_INPUT_CURRENT_LIMIT,
+			       &input_current_limit) < 0) {
+		printk("Can't read LTC4162 input current limit\n");
+	}
+
+	ltc4162_show_values("Input current limit in Amps", input_current_limit);
+
+	if (sensor_sample_fetch_chan(dev,
+			SENSOR_CHAN_CHARGER_TEMPERATURE) < 0) {
+		printk("Failed to fetch charger temperature\n");
+		return;
+	}
+
+	if (sensor_channel_get(dev, SENSOR_CHAN_CHARGER_TEMPERATURE,
+			       &charger_temperature) < 0) {
+		printk("Can't read LTC4162 Charger temperature\n");
+		return;
+	}
+
+	ltc4162_show_values("Charger die temperature in Celsius", charger_temperature);
+
+	printk("\n");
+}
+
+void main(void)
+{
+	int ret;
+	struct sensor_value chrg_prop;
+	const struct device *dev = DEVICE_DT_GET_ANY(adi_ltc4162);
+
+	if (dev == NULL) {
+		printk("Failed to get device\n");
+		return;
+	}
```

Comment:
`DEVICE_DT_GET_ONE` can be used to obtain a build failure. cc: @mbolivar-nordic we should likely settle on `DEVICE_DT_GET_ONE` :-)

Reply:
using `DEVICE_DT_GET_ANY` macro to get the device